app/build.gradle

```diff
@@ -576,12 +576,12 @@ dependencies {
     hvr3dofImplementation 'com.huawei.agconnect:agconnect-core-harmony:1.1.0.300'
     hvrImplementation 'com.huawei.agconnect:agconnect-core:1.6.0.300'
     hvr3dofImplementation 'com.huawei.agconnect:agconnect-core:1.6.0.300'
-    hvrImplementation 'com.huawei.hms:hianalytics:6.3.2.300'
-    hvr3dofImplementation 'com.huawei.hms:hianalytics:6.3.2.300'
-    hvrImplementation 'com.huawei.hms:ml-computer-voice-asr:3.1.0.300'
-    hvr3dofImplementation 'com.huawei.hms:ml-computer-voice-asr:3.1.0.300'
-    hvrImplementation 'com.huawei.hms:location:6.2.0.300'
-    hvr3dofImplementation 'com.huawei.hms:location:6.2.0.300'
+    hvrImplementation 'com.huawei.hms:hianalytics:6.5.0.300'
+    hvr3dofImplementation 'com.huawei.hms:hianalytics:6.5.0.300'
+    hvrImplementation 'com.huawei.hms:ml-computer-voice-asr:3.5.0.301'
+    hvr3dofImplementation 'com.huawei.hms:ml-computer-voice-asr:3.5.0.301'
+    hvrImplementation 'com.huawei.hms:location:6.4.0.300'
+    hvr3dofImplementation 'com.huawei.hms:location:6.4.0.300'
```

Comment:
Are these changes strictly required?

You're using 6.4 for location services. Is that correct? I just ask because you're using 6.5 for the others.

Reply:
Moved to a separate PR.